WordPress/gutenberg
# TextControl: Convert component to TypeScript

## What?
Converts the `TextControl` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
Refactors the current `TextControl` to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `TextControl` continues to function as expected
3. Test the various text controls within the block/site editors in Gutenberg still work

cc: @ciampo
